src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -1986,6 +1988,9 @@ public TypeVar(TypeSymbol tsym, Type bound, Type lower,
             this.lower = lower;
             this.isReferenceProjection = isReferenceProjection;
             this.isUniversal = (tsym.flags_field & UNIVERSAL) != 0;
+            if (isUniversal && !isReferenceProjection) {
```

Comment:
isn't `isReferenceProjection` always `false` here (you have just created the class)

Reply:
no, because this constructor is used again to build the reference, so I have to add it to the condition to avoid stackoverflow errors